buffer/src/test/java/io/netty/buffer/RetainedSlicedByteBufTest.java

```diff
@@ -26,8 +25,7 @@
     protected ByteBuf newBuffer(int length, int maxCapacity) {
         Assume.assumeTrue(maxCapacity == Integer.MAX_VALUE);
         ByteBuf wrapped = Unpooled.wrappedBuffer(new byte[length * 2]);
-        ByteBuf buffer = wrapped.retainedSlice(length > 1 ?
-                        PlatformDependent.threadLocalRandom().nextInt(length - 1) + 1 : 0, length);
+        ByteBuf buffer = wrapped.retainedSlice(length > 1 ? 1 : 0, length);
```

Comment:
Would this give us more variability in the test while still ensuring we have at least `length` bytes?
```java
start = /* random number from 0 to length * 2 - length */
ByteBuf buffer = wrapped.retainedSlice(start, length);
```